### ISSUE TITLE
soroban-rpc: [spike] Datastore choice for rpc backend

### ISSUE THREAD

**Issue by mollykarcher** [stellar/go#4731]
The [backend for the rpc](https://docs.google.com/document/d/1COLQe2BpHFMCj3SR25qi3PyuehXHraqN05T_F3kv6D4/edit#) will need to support both ledger entries and events. We'd like to use the same datastore for both, and would like to do some performance and viability testing of the possible options before committing to the implementation. 

Draw on learnings from prior [investigation](https://docs.google.com/document/d/1F60xINIbfKrdnGeBBxygeHU3ZlgthRJw8DLXaYDakl0/edit) already done while assuming events was the only data we were storing.

Once the proof-of-concept for this is complete, the implementation for both use cases will be done in the following related issues:
- #4717 
- #4718
